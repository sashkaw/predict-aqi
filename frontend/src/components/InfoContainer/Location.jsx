import { useAQIContext } from "../../contexts/AirQualityContext";
import sites from "../../monitoring_sites.json";

export function Location() {
    const { setSite } = useAQIContext();

    function handleChange(event) {
        const selectedSite = sites.filter(
            (site) => site.id == Number(event.target.value)
        )[0];
        setSite(selectedSite);
    }

    return (
        <div className="location-container">
            <select 
                defaultValue={"1"}
                onChange={handleChange}
                className="location-button"
            >
                {sites.map((site) => {
                    return (
                        <option value={site.id} key={site.id}>
                            {site.name}
                        </option>
                    );
                })}
            </select>
        </div>
    );
}
