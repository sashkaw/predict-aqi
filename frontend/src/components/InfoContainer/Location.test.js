import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { useEffect, useState, useContext } from "react";
import TestAQIProvider from "../Fixtures";
import { Location } from "./Location";

// Set up Location with mock data
const Fixture = () => {
    const Child = () => {
        return(
            <Location />
        )
    };
    // Return Location wrapped within Provider
    return (
        <TestAQIProvider>
            <Child />
        </TestAQIProvider>
    );
}
// Helper function for snapshot testing
const matches = (children) => expect(
    renderer.create(children).toJSON()
).toMatchSnapshot();

describe("<Location />", () => {
    test("Create location container", () => {
        matches(<Fixture />);
    })
})

//TODO: add tests for handleChange


