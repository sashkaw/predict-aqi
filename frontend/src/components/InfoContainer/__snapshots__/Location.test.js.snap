// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Location /> Create location container 1`] = `
<div
  className="location-container"
>
  <select
    className="location-button"
    defaultValue="1"
    onChange={[Function]}
  >
    <option
      value={1}
    >
      Amazon Park - Eugene, OR
    </option>
  </select>
</div>
`;
